### PR TITLE
User can comment on another post

### PR DESCRIPTION
**Pull request by salindholm**
[User can comment on another post](https://www.pivotaltracker.com/story/show/176728915)

## User Story
```
As a User,            
In order to voice my opinions,         
I want to be able to leave a comment.
```

## Changes proposed in this pull request:
* Updating comment service module with post request
* Changing naming standard in the initial state to match API
* Adding several hooks to comment section for use in onClick event
* Adding clear button to comment input

## What I have learned working on this feature:
* How to work with Route params

## Screenshots (Client)
Screenshot from 2021-02-02 13-10-29

### CODE REVIEW

File: src/components/CommentSection.jsx [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
-      <Text testID='single-comment'>{content}</Text>
-    </View>
+  useEffect(() => {
+    Comments.index(post.id)
+  }, [])
+
+
+  const SingleComment = ({ content }) => <Text>{content}</Text>
+  const renderComment = ({ item }) => (
+    <SingleComment testID={`comment-text-${item.id}`} content={item.content} />
```
Comment (ViktorHek) [inline src/components/CommentSection.jsx]: you don't need this testID any more

File: src/components/PostIndex.jsx [hunk 2]
```diff
@@ -33,10 +33,11 @@ const PostIndex = ({ post }) => {
         <Card.Divider />
         <TouchableOpacity
           style={styles.commentButton}
+          title='Comment'
           testID='comment-button'
           onPress={() => navigation.navigate('CommentSection', { post: post })}
         >
-          Comment
+          <Text>Comment</Text>
```
Comment (ViktorHek) [inline src/components/PostIndex.jsx]: we don't need the title if we have the `<Text>`

File: src/components/CommentSection.jsx (hunk 1 above)
Comment (salindholm) [reply src/components/CommentSection.jsx]: Okey, removed!

File: src/components/PostIndex.jsx (hunk 2 above)
Comment (salindholm) [reply src/components/PostIndex.jsx]: Good point!

File: src/components/CommentSection.jsx [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
   )
-  const renderComment = ({ item }) => <SingleComment content={item.content} />
 
   return (
     <View testID='comment-section' name='CommentSection'>
       <FlatList
         data={comments}
         keyExtractor={item => item.id.toString()}
         renderItem={renderComment}
+        extraData={newComment}
```
Comment (KyunginNa) [inline src/components/CommentSection.jsx]: We might don't need this. Can you check if the app is working without extraData?

Comment (salindholm) [reply src/components/CommentSection.jsx]: It works without, we have now removed it!

**Pull request merged**